src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6600,4 +6600,40 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_FunctionPointerTypesInAttributeNotSupported" xml:space="preserve">
     <value>Using a function pointer type in a 'typeof' in an attribute is not supported.</value>
   </data>
-</root>
+  <data name="ERR_BadCallerArgumentExpressionParamWithoutDefaultValue" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute may only be applied to parameters with default values</value>
+  </data>
+  <data name="ERR_NoConversionForCallerArgumentExpressionParam" xml:space="preserve">
+    <value>CallerArgumentExpressionAttribute cannot be applied because there are no standard conversions from type '{0}' to type '{1}'</value>
```

Comment:
>standard [](start = 84, length = 8)

Should conversio be also implicit?

Reply:
I copied the exact same text from existing diagnostics. I'm not entirely sure about the correct terminology. Will need to go through the [Standard conversions](https://github.com/dotnet/csharplang/blob/main/spec/conversions.md#standard-conversions) in spec.

Reply:
@AlekseyTs The following code compiles (standard *explicit* conversion from `object` to `string` - based on my understanding from spec):

```csharp
using System.Runtime.CompilerServices;

public class C
{
    public void M([CallerMemberName] object name = null)
    {
    }
}
```

So it seems standard conversion is correct

Reply:
@AlekseyTs Consider the following code:

```csharp
using System.Runtime.CompilerServices;

public class C
{
    public static void M([CallerMemberName] C name = null)
    {
    }
}
```

The error message is:

> error CS4019: CallerMemberNameAttribute cannot be applied because there are no standard conversions from type 'string' to type 'C'

Changing `C` to `object` compiles fine. So it looks like these kind of attribute requires a standard conversion from `string` to whatever the declared type is.

Reply:
> Changing C to object compiles fine. So it looks like these kind of attribute requires a standard conversion from string to whatever the declared type is.

The question is whether the standard conversion should also be implicit.

---
In reply to: [600628537](https://github.com/dotnet/roslyn/pull/51952#discussion_r600628537) [](ancestors = 600628537)

Reply:
@AlekseyTs, Well, looking at the [spec of CallerMemberName](https://github.com/dotnet/csharplang/blob/1b4f6ac09fd0f29593cc8bf96244e2b16e2509cd/spec/attributes.md#the-callermembername-attribute):

> The System.Runtime.CompilerServices.CallerMemberNameAttribute is allowed on optional parameters when there is a standard implicit conversion (Standard implicit conversions) from string to the parameter's type.

However, the compiler allows the parameter of a CallerMemberName to be `object`. I don't think the conversion from `string` to `object`. So it seems like the current implementation violates the spec. (unless there is something I'm not understanding).

Reply:
Ops I was swapping the source and destination for conversion in mind :smile:
Yes it should be standard implicit conversion.